src/plugins/translate/TranslationManager.js

```diff
@@ -67,54 +71,43 @@ export class TranslationManager {
 
   async initWorker() {
     if (this.initPromise) return this.initPromise;
-
-    if (window.Worker) {
-      this.worker = new Worker("/BookReader/translate/worker.js");
-      this.worker.postMessage(["import"]);
-    }
-
     this.initPromise = new Promise((resolve, reject) => {
       this._initResolve = resolve;
       this._initReject = reject;
     });
+    const registryJson = await fetch("https://cors.archive.org/cors/mozilla-translate-models/registry.json").then(r => r.json());
+    for (const language of Object.values(registryJson)) {
+      for (const file of Object.values(language)) {
+        file.name = "https://cors.archive.org/cors/mozilla-translate-models/" + file.name;
```

Comment:
Let's update this to use the path of the registry url, eg
```suggestion
        file.name = registryUrl.replace('/[^\/]+$/', '/') + file.name;
```

That will make all the file names relative to the given url.